std/assembly/typedarray.ts

```diff
@@ -1734,27 +1734,26 @@ function REVERSE<TArray extends ArrayBufferView, T>(array: TArray): TArray {
 // @ts-ignore: decorator
 @inline
 function WRAP<TArray extends ArrayBufferView, T>(buffer: ArrayBuffer, byteOffset: i32 = 0, length: i32 = -1): TArray {
+  var byteLength: i32;
   var bufferByteLength = buffer.byteLength;
-  if (<u32>byteOffset >= <u32>bufferByteLength) {
+  // @ts-ignore: cast
+  if ((<u32>byteOffset > <u32>bufferByteLength) | (byteOffset & (sizeof<T>() - 1))) {
     throw new RangeError(E_INDEXOUTOFRANGE);
   }
-  var byteLength: i32;
   if (length < 0) {
     if (length == -1) {
-      const mask = <i32>(1 << alignof<T>() - 1);
-      if (buffer.byteLength & mask) {
+      if (bufferByteLength & (sizeof<T>() - 1)) {
         throw new RangeError(E_INVALIDLENGTH);
-      } else {
-        byteLength = buffer.byteLength;
       }
+      byteLength = bufferByteLength - byteOffset;
     } else {
       throw new RangeError(E_INVALIDLENGTH);
     }
   } else {
     byteLength = length << alignof<T>();
-  }
-  if (byteOffset + byteLength > buffer.byteLength) {
-    throw new RangeError(E_INVALIDLENGTH);
+    if (byteOffset + byteLength > bufferByteLength) {
```

Comment:
In this case the code does not check that `buferByteLength` is aligned, is that correct?

Reply:
Yes. According to spec it's not needed
https://tc39.es/ecma262/#sec-typedarray-buffer-byteoffset-length